src/Build/BackEnd/Components/SdkResolution/SdkResult.cs

```diff
@@ -52,11 +55,15 @@ public SdkResult()
 
         public override string Version => _version;
 
+        public override bool TrackEnvironmentVariables { get => _trackEnvironmentVariables; set => _trackEnvironmentVariables = value; }
```

Comment:
Seems superfluous to have resolvers dynamically respond whether they track environment variables on every sdk resolution. Probably better to ask them once, during resolver loading in Microsoft.Build.BackEnd.SdkResolution.SdkResolverLoader

Like SdkResolver.GetCapabilities, and the env tracking would be part of a return enum.